procedures/CodeBrowser.ipf

```diff
@@ -878,19 +916,52 @@ Function/S nicifyProcedureList(list)
 	return niceList
 End
 
+// Get the specified line of code from a function
+//
+// @param funcInfo  output of FunctionInfo for the function in question
+// @param lineNo    line number relative to the function definition
+//                  set to -1 to return lines before the procedure that are not part of the preceding macro or function
+//                  see `DisplayHelpTopic("ProcedureText")`
+//
+// @returns lines of code from a function inside a procedure file
+Function/S getFunctionLine(lineNo, funcInfo)
+	variable lineNo
+	string funcInfo
+
+	string funcName, module, procedure, context
+	variable linesOfContext
+
+	funcName = StringByKey("NAME", funcInfo)
+	module = StringByKey("INDEPENDENTMODULE", funcInfo)
+	procedure = StringByKey("PROCWIN", funcInfo)
+
+	if(!isProcGlobal(module))
+		debugPrint(procedure + " is not in ProcGlobal")
+		procedure = procedure + "[" + module + "]"
+	endif
+
+	linesOfContext = lineNo < 0 ? lineNo : 0
+	context = ProcedureText(funcName, linesOfContext, procedure)
```

Comment:
You should reuse `getProcedureText` here which then needs an optional linesOfContext parameter. In that way you don't need to do the module handling here as well.